src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/GpNameAndAddress.java

```diff
@@ -0,0 +1,59 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.EdifactValidationException;
+
+import org.springframework.util.StringUtils;
+
+/**
+ * Example NAD+GP+2750922,295:900'
+ */
+@Getter @Setter @RequiredArgsConstructor
+public class GpNameAndAddress extends Segment{
+
+    public static final String KEY = "NAD";
+    public static final String QUALIFIER = "GP";
+    public static final String KEY_QUALIFIER = KEY + "+" + QUALIFIER;
+    private @NonNull String identifier;
+    private @NonNull String code;
+
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        return QUALIFIER + "+" + identifier + ":" + code;
+    }
+
+    @Override
+    protected void validateStateful() {
+        //NOP
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        if (StringUtils.isEmpty(identifier)) {
+            throw new EdifactValidationException(getKey() + ": Attribute identifier is required");
+        }
+        //Is below needed as it will always be populated
```

Comment:
Will it? `new GpNameAndAddress("", "")` is possible right?

Reply:
ahhh copy-paste error :)